### PR TITLE
facts: Update RAID devices regex for FreeBSD

### PR DESCRIPTION
**Pull request by lostsoulparty**
##### SUMMARY

Added support FreeBSD RAID devices and regex to match partitions as well as slices.  
RAID device list taken from here: 
https://github.com/freebsd/freebsd-src/blob/main/usr.sbin/bsdconfig/share/device.subr

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

There's a demand to gather information of RAID devices and partitions.

### CODE REVIEW

**Comment by jborean93** [comment]
Thanks for your contribution, we would need to see a changelog fragment as documented under https://docs.ansible.com/ansible/latest/community/development_process.html#creating-changelog-fragments. It would also be great to see some mock fact results that test out this new regex pattern as well as the sanity tests passing. If you need help with this I encourage you to reach out over IRC in the #ansible-devel channel

* IRC: #ansible on [irc.libera.chat](https://libera.chat/)

File: lib/ansible/module_utils/facts/hardware/freebsd.py [hunk 1]
```diff
@@ -173,8 +173,11 @@ def get_device_facts(self):
 
         sysdir = '/dev'
         device_facts['devices'] = {}
-        drives = re.compile(r'(ada?\d+|da\d+|a?cd\d+)')  # TODO: rc, disks, err = self.module.run_command("/sbin/sysctl kern.disks")
-        slices = re.compile(r'(ada?\d+s\d+\w*|da\d+s\d+\w*)')
+        # TODO: rc, disks, err = self.module.run_command("/sbin/sysctl kern.disks")
+        drives = re.compile(r'(ada?\d+|da\d+|a?cd\d+|amrd\d+|da\d+|idad\d+|ipsd\d+|md\d+|mfid\d+|mlxd\d+|twed\d+|vtbd\d+)')
+        slices = re.compile(
+            (r'(ada?\d+[ps]\d+\w*|a?cd\d+[ps]\d+\w*|amrd\d+[ps]\d+\w*|da\d+[ps]\d+\w*|idad\d+[ps]\d+\w*|ipsd\d+[ps]\d+\w*'
+             r'|md\d+[ps]\d+\w*|mfid\d+[ps]\d+\w*|mlxd\d+[ps]\d+\w*|twed\d+[ps]\d+\w*|vtbd\d+[ps]\d+\w*)'))
```
Comment (webknjaz) [inline lib/ansible/module_utils/facts/hardware/freebsd.py]: Readability tip: It is possible to use multiline raw-strings for and spread the regex across lines — https://docs.python.org/3/library/re.html#re.VERBOSE. As in
```python
r"""(?x)
\w+  # comment 1
\s+
\d+  # comment 2
"""
```

Comment (Akasurde) [reply lib/ansible/module_utils/facts/hardware/freebsd.py]: @lostsoulparty Any updates?

File: lib/ansible/module_utils/facts/hardware/freebsd.py [hunk 2]
```diff
@@ -173,8 +173,41 @@ def get_device_facts(self):
...
+                a?cd\d+[ps]\d+\w*|  # SCSI CDROM drive
+                amrd\d+[ps]\d+\w*|  # AMI MegaRAID drive
+                da\d+[ps]\d+\w*|    # SCSI disk device
+                idad\d+[ps]\d+\w*|  # Compaq RAID array
+                ipsd\d+[ps]\d+\w*|  # IBM ServeRAID RAID array
+                md\d+[ps]\d+\w*|    # md(4) disk device
+                mfid\d+[ps]\d+\w*|  # LSI MegaRAID SAS array
+                mlxd\d+[ps]\d+\w*|  # Mylex RAID disk
+                twed\d+[ps]\d+\w*|  # 3ware ATA RAID array
+                vtbd\d+[ps]\d+\w*   # VirtIO Block Device
```
Comment (webknjaz) [inline lib/ansible/module_utils/facts/hardware/freebsd.py]: Looks like `re.VERBOSE` flag (or it's `(?x)` variant) is missing. Without it, multi-lining wouldn't work. Perhaps, a test would be useful to catch such things..

Additionally, the variable got renamed accidentally (with the trailing `s` being dropped) and the control flow errors out elsewhere.

Finally, I noticed that there's a lot of duplication that can hopefully be addressed as well, with the following patch:

```suggestion
            r"""(?x)(
              (?:
                ada?   # ATA/SATA disk device
                |da    # SCSI disk device
                |a?cd  # SCSI CDROM drive
                |amrd  # AMI MegaRAID drive
                |idad  # Compaq RAID array
                |ipsd  # IBM ServeRAID RAID array
                |md    # md(4) disk device
                |mfid  # LSI MegaRAID SAS array
                |mlxd  # Mylex RAID disk
                |twed  # 3ware ATA RAID array
                |vtbd  # VirtIO Block Device
              )\d+
            )
            """
        )

        slice = re.compile(
            r"""(?x)(
              (?:
                ada?   # ATA/SATA disk device
                |a?cd  # SCSI CDROM drive
                |amrd  # AMI MegaRAID drive
                |da    # SCSI disk device
                |idad  # Compaq RAID array
                |ipsd  # IBM ServeRAID RAID array
                |md    # md(4) disk device
                |mfid  # LSI MegaRAID SAS array
                |mlxd  # Mylex RAID disk
                |twed  # 3ware ATA RAID array
                |vtbd  # VirtIO Block Device
              )\d+[ps]\d+\w*
```

*(I haven't tested this so there may be typos)*

**Review by bcoca — APPROVED** [review]

**Pull request merged**